sway/commands/output.c

```diff
@@ -62,8 +62,30 @@ struct cmd_results *cmd_output(int argc, char **argv) {
 			output_get_identifier(identifier, 128, sway_output);
 			output = new_output_config(identifier);
 		}
+		if (sway_output->enabled && !sway_output->wlr_output->enabled) {
+			output->dpms_state = DPMS_OFF;
+		} else {
+			output->dpms_state = DPMS_ON;
+		}
 	} else {
 		output = new_output_config(argv[0]);
+
+		// Obtain current DPMS state for the specified output,
+		// on wildcard just choose focused output
+		struct sway_output *sway_output = NULL;
+		if (strcmp(argv[0], "*") == 0) {
```

Comment:
This would modify the output dpms state for non-dpms wildcard output commands too though.

I don't think we should modify the output config outside of the handler, so all this logic should probably go into output/dpms.c.

Reply:
Oh, I missed that bit, thanks. Gonna move it inside the handler.

Thought it'd be a bit cleaner to put this logic here. But now, looking at output/toggle.c, it makes a lot more sense to just stay inside the handler.